src/middy.js

```diff
@@ -72,7 +72,7 @@ const runErrorMiddlewares = (middlewares, instance, done) => {
   const runNext = (err) => {
     try {
       if (!err) {
-        return done()
+        instance.__handledError = true
```

Comment:
Just updated the code @peterjcaulfield, @dkatavic! Please have  another look when you can.